packages/common/test/tools/EventTracker.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from 'earljs'
+import { setTimeout } from 'timers/promises'
+
+import { EventTracker } from '../../src/tools/EventTracker'
+
+describe(EventTracker.name, () => {
+  it('has empty history by default', () => {
+    const tracker = new EventTracker()
+    expect(tracker.getStats()).toEqual({
+      lastSecond: {},
+      lastMinuteAverage: {},
+      lastHourAverage: {},
+    })
+  })
+
+  it('records event', () => {
+    const tracker = new EventTracker()
+    tracker.record('something')
+    expect(tracker.getStats()).toEqual({
+      lastSecond: {
+        something: 1,
+      },
+      lastMinuteAverage: {
+        something: 1 / 60,
+      },
+      lastHourAverage: {
+        something: 1 / 3_600,
+      },
+    })
+  })
+
+  it('groups event', async () => {
+    const tracker = new EventTracker()
```

Comment:
don't we want to mock timers in tests? i am pretty sure we already do it across the codebase